packages/react-native/src/components/3DS/useThreeDSecure.tsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from "react";
+import { useRef } from "react";
+import { useEvervault } from "../EvervaultProvider";
+import { startSession, threeDSecureSession } from "./threeDSSession";
+import {
+  ThreeDSecureCallbacks,
+  UseThreeDSecureResponse,
+  UseThreeDSecureState,
+} from "./types";
+
+export const useThreeDSecure = (): UseThreeDSecureResponse => {
+  const { appUuid } = useEvervault();
+  const intervalRef = useRef<NodeJS.Timeout | null>(null);
+  const [displayModal, setDisplayModal] = useState(false);
+  const [state, setState] = useState<UseThreeDSecureState>({
+    session: null,
+    displayModal,
+  });
+
+  useEffect(() => {
+    setState((prevState) => ({
+      ...prevState,
+      displayModal,
+    }));
+  }, [displayModal]);
```

Comment:
This can be removed if we separate the state like described in my previous comment.